contrib/babelfishpg_tds/src/backend/tds/tdsutils.c

```diff
@@ -1148,3 +1186,113 @@ is_babelfish_ownership_enabled(ArrayType *array)
 	}
 	return false;
 }
+
+/*
+ * handle_alter_role
+ *
+ * Description: This function handles dealing with ALTER ROLE <role> WITH.
+ *
+ * Returns: true - We're not attempting to modify something we shouldn't have access to. Normal security checks.
+ *          false - We've reported an error and should not continue executing this call.
+ */
+static bool
+handle_alter_role(AlterRoleStmt* alter_role_stmt)
+{
+	List *options = alter_role_stmt->options;
+	ListCell *opt;
+	char *name = get_role_name(alter_role_stmt->role);
+	bool allow_alter_role_operation = enable_alter_babelfish_role;
+
+	/*
+	 * If the role does not exist, just let the normal Postgres checks happen.
+	 * If the role is "sysadmin" then allow the alter operation.
+	 */
+	if (name == NULL || (strcmp(name, BABELFISH_SYSADMIN) == 0))
+	{
+		return true;
+	}
+
+	/*
+	 * If the enable_alter_babelfish_role guc variable is set to true and tried to execute
+	 * ALTER ROLE ... WITH statement(other than password change, connection limit, valid until),
+	 * then change the value of allow_alter_role_operation variable to false as we disallow
+	 * alter operation[WITH statement](other than password change, connection limit, valid until) from PG end.
+	 */
+	if(allow_alter_role_operation)
+	{
+		/*
+		 * There are only few operations which need to be allowed for ALTER ROLE <role> WITH namely
+		 * password change, connection limit, valid until.
+		 *
+		 * If any non-allowed option is given among the options then disallow the operation.
+		 *
+		 * In fututre, if need to allow more operations then add those options into the list.
+		 */
+		foreach(opt, options)
+		{
+			DefElem *defel = (DefElem *) lfirst(opt);
+			if (strcmp(defel->defname, "password") == 0 ||
+				strcmp(defel->defname, "connectionlimit") == 0 ||
+				strcmp(defel->defname, "validUntil") == 0)
+					allow_alter_role_operation = true;
+			else
+			{
+				allow_alter_role_operation = false;
+				break;
+			}
+		}
+	}
+	check_babelfish_alterrole_restictions(name, allow_alter_role_operation);
+
+	/* We don't need "name" anymore */
+	pfree(name);
+	return true;
+}
+
+/* handle_alter_role_set
+ *
+ * Description: This function handles dealing with ALTER ROLE <role> SET.
+ *
+ * Returns: true - We're not attempting to modify something we shouldn't have access to, continue on.
+ *          false - We've reported an error and should not continue executing this call.
+ */
+static bool
+handle_alter_role_set (AlterRoleSetStmt* alter_role_set_stmt)
```

Comment:
ALTER ROLE { role_specification | ALL } [ IN DATABASE database_name ]

According the doc, it could be all roles in one database. 
So for BBF's case, it could alter all roles in babelfish_db. Please consider this as well.